### PR TITLE
[ENH] native grid search in TSCGridSearchCV and TSRGridSearchCV via evaluate

### LINKED ISSUE #10724: [ENH] in `TSCGridSearchCV` and `TSRGridSearchCV`, replace use of `sklearn` grid search by `sktime` native `evaluate` call
Currently, `TSCGridSearchCV` and `TSRGridSearchCV` use the `sklearn` `GridSearchCV` internally.

`scikit-learn` have changed their API, and this will no longer be possible in future versions due to an incompatibility in the tag system.

Hence the internal logic should be replaced by a use of `model_selection.evaluate` from the `classification` resp `regression` module, similar to how `ForecastingGridSearchCV` is working.

`TSCOptCV` already provides non-`sklearn` grid search - and one pathway would be minimal vendoring, or delegation.

However, I would prefer removing the abstraction layer and exposing only the grid search related functionality, i.e., no `hyperactive` vendoring.

### PR DESCRIPTION
**Pull request by atikulmunna**
*LLM generated content, by Claude Fable 5; reviewed, tested, and submitted by the author.*

#### Reference Issues/PRs

Fixes #10724. Progress on #6708.

#### What does this implement/fix? Explain your changes.

`TSCGridSearchCV` and `TSRGridSearchCV` were thin wrappers around sklearn's `GridSearchCV`, which breaks with upcoming scikit-learn versions due to the tag system incompatibility. This rewrites both tuners to run grid search natively, following the design of `ForecastingGridSearchCV`: candidates come from `ParameterGrid`, each candidate is backtested via the shared `_evaluate` from `classification.model_evaluation`, parallelized via `utils.parallel.parallelize` with new `backend` and `backend_params` parameters. The delegate is now `best_estimator_`. No `hyperactive` vendoring.

Behavior changes a reviewer should be aware of:

- `cv_results_` is now a `pd.DataFrame` with `mean_test_<metric_name>`, `mean_fit_time`, `mean_pred_time`, `params`, and `rank_test_<metric_name>` columns, consistent with the forecasting tuners, instead of the sklearn dict format.
- `scoring` accepts callables or lists of callables. sklearn scorer strings now raise an informative `TypeError`.
- Metric direction is inferred from the metric name: `*_error`, `*_loss`, `*_deviance`, `*_risk` are treated as lower-is-better. Default scoring stays `accuracy_score` for classification and is now `mean_squared_error` for regression, the same default as `evaluate`.
- `cv` int/None resolve to deterministic `KFold` (3 splits for None), so all candidates are evaluated on identical splits. No stratification, since `evaluate` splits without passing `y`.
- `pre_dispatch` and `return_train_score` are retained in the signature for backward compatibility but are inert. `n_jobs` maps to the `loky` backend.
- `refit=False` now raises an informative `RuntimeError` on predict, instead of sklearn's `NotFittedError`.

This also fixes an inverted vectorization assertion in `test_multioutput` in `test_all_regressors.py` (the classifier counterpart has the correct `not`), and removes the `TSRGridSearchCV` exclusion from `tests/_config.py`, since the estimator now passes that test.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

- whether the behavior changes above are acceptable, in particular the `cv_results_` format and the metric direction heuristic
- whether deprecation warnings are wanted for `pre_dispatch`, `return_train_score`, and `n_jobs` instead of the silent backward compatible treatment

#### Did you add any tests for the change?

Yes. Dedicated test modules for both tuners, which did not exist before: scores are checked against manual `evaluate` runs per candidate, ranking direction for lower-is-better metrics, multi-metric scoring, `refit=False` behavior, and the string scoring error. Both classes pass `check_estimator` locally, 256 checks for TSC and 144 for TSR, including the previously excluded multioutput test.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) via the all-contributors file.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].

### CODE REVIEW

**Comment by atikulmunna** [comment]
CI has not run on this yet. It needs a maintainer to approve the workflow, since this is my first PR here. Could someone kick it off?

Current state after a rebase on main just now, so the branch is conflict free:

- `check_estimator` green for both classes, 257 checks for `TSCGridSearchCV` and 145 for `TSRGridSearchCV`, 0 failures
- 10 new tests in `classification/model_selection/tests/test_tune.py` and `regression/model_selection/tests/test_tune.py` pass
- `test_multioutput` passes for both `TSRGridSearchCV` instances, which is what let me drop the exclusion in `tests/_config.py`
- class doctests run clean for both

Happy to split this into smaller PRs if the size is the blocker.

File: sktime/regression/tests/test_all_regressors.py [hunk 1]
```diff
@@ -106,7 +106,8 @@ def test_multioutput(self, object_instance):
         assert isinstance(y_pred, pd.DataFrame)
         assert y_pred.shape == y_mult.shape
 
-        vectorized = object_instance.get_tag("capability:multioutput")
+        # the estimator vectorizes iff it does not have the multioutput capability
+        vectorized = not object_instance.get_tag("capability:multioutput")
```
Comment (fkiraly) [inline sktime/regression/tests/test_all_regressors.py:110]: this looks like a bug, can you please fix this in a separate PR?

**Review by fkiraly — CHANGES_REQUESTED** [review]
This looks like a lot of copy-paste, instead of using `evaluate` as requested.
